pyomo/contrib/fbbt/tests/test_fbbt.py

```diff
@@ -1045,6 +1050,37 @@ def test_fixed_var(self):
         self.assertAlmostEqual(m.y.lb, 1)
         self.assertAlmostEqual(m.y.ub, 2)
 
+    @unittest.skipUnless(
+        flib, 'Could not find the "asl_external_demo.so" library')
+    @unittest.skipIf(is_pypy, 'Cannot evaluate external functions under pypy')
+    def test_external_function(self):
+        if self.tightener is not fbbt:
+            raise unittest.SkipTest('Appsi FBBT does not support unkown expressions yet')
```

Comment:
Should we add a ticket to remind ourselves to add `ExternalFunctionExpression` to APPSI?  I believe the issue is not actually the `ExternalFunctionExpression`, but rather that they are allowed (by AMPL and Pyomo) to have `str` arguments.

Reply:
I'm fairly certain an exception would be raised somewhere else if we fixed the `str` problem. Yes, we should add a ticket.